sharding-spring/sharding-jdbc-orchestration-spring/sharding-jdbc-orchestration-spring-namespace/src/main/java/org/apache/shardingsphere/shardingjdbc/orchestration/spring/namespace/parser/DataSourceBeanDefinitionParser.java

```diff
@@ -66,10 +71,17 @@ private void configureFactory(final Element element, final BeanDefinitionBuilder
     
     private BeanDefinition getOrchestrationConfiguration(final Element element) {
         BeanDefinitionBuilder factory = BeanDefinitionBuilder.rootBeanDefinition(OrchestrationConfiguration.class);
-        factory.addConstructorArgValue(element.getAttribute(ID_ATTRIBUTE));
-        factory.addConstructorArgReference(element.getAttribute(ShardingDataSourceBeanDefinitionParserTag.REG_REF_TAG));
-        factory.addConstructorArgValue(element.getAttribute(ShardingDataSourceBeanDefinitionParserTag.OVERWRITE_TAG));
+        factory.addConstructorArgValue(parseInstances(element));
         return factory.getBeanDefinition();
     }
+    
+    private Map<String, RuntimeBeanReference> parseInstances(final Element element) {
+        List<String> instances = Splitter.on(",").trimResults().splitToList(element.getAttribute(ShardingDataSourceBeanDefinitionParserTag.INSTANCE_REF_TAG));
+        Map<String, RuntimeBeanReference> result = new ManagedMap<>(instances.size());
```

Comment:
I suggest use Guava's `Maps` instead of `ManagedMap`.

Reply:
`ManagedMap` is different with usual `HashMap`, it can hold managed Map values, which may
include runtime bean references (to be resolved into bean objects).